remotes/docker/resolver.go

```diff
@@ -291,10 +291,14 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			resp.Body.Close() // don't care about body contents.
 
 			if resp.StatusCode > 299 {
-				if resp.StatusCode == http.StatusNotFound {
-					continue
+				if resp.StatusCode > 399 {
+					// Set lastErr when encountering the first non-404 status code.
+					if lastErr == nil && resp.StatusCode != http.StatusNotFound {
```

Comment:
nit: https://github.com/containerd/containerd/pull/4575#issuecomment-695836271 should be commented somewhere.

Reply:
Done